runbotics/runbotics-scheduler/src/scheduler-database/tenant/tenant.controller.ts

```diff
@@ -0,0 +1,143 @@
+        this.logger.log('REST request to create invite code for user tenant with id: ', id);
+
+        if (await this.tenantService.getActiveInviteCodeByTenantId(id)) {
+            this.logger.error('Valid code exists for tenant with id: ', id);
+            throw new BadRequestException('Valid code exists for tenant');
+        }
+
+        return this.tenantService.createInviteCodeByTenantId(id);
+    }
+
+    // -----------------------------------
+
+    @Get() // TODO: pagination & filtering
+    @FeatureKeys(FeatureKey.TENANT_ALL_ACCESS)
+    getAllTenants() {
+        this.logger.log('REST request to get all tenants');
+        return this.tenantService.getAllTenants();
+    }
+
+    @Get('/invite-code/:tenantId')
+    @FeatureKeys(FeatureKey.TENANT_GET_ALL_INVITE_CODE)
+    async getActiveInviteCodeByTenant(@Param('tenantId') id: Tenant['id']) {
+        this.logger.log(`REST request to get tenant invite code for tenant with id: ${id}`);
+        const inviteCodeDto = await this.tenantService.getActiveInviteCodeByTenantId(id);
+
+        if (!inviteCodeDto) {
+            this.logger.error('Cannot find valid invite code for tenant with id: ', id);
+            throw new NotFoundException('Invite code not found');
+        }
+
+        return inviteCodeDto;
+    }
+
+    @Get('/:id')
+    @FeatureKeys(FeatureKey.TENANT_ALL_ACCESS)
+    async getTenantById(
+        @Param('id', ParseUUIDPipe) id: Tenant['id']
+    ) {
+        this.logger.log('REST request to get tenant by id: ', id);
+        const tenant = await this.tenantService.getTenantById(id);
+
+        if (!tenant) {
+            this.logger.error('Cannot find tenant with id: ', id);
+            throw new NotFoundException('Tenant not found');
+        }
+
+        return tenant;
+    }
+
+    @Post()
+    @FeatureKeys(FeatureKey.TENANT_ALL_ACCESS)
+    createTenant(
+        @Body(new ZodValidationPipe(createTenantSchema)) tenantDto: CreateTenantDto,
+        @User() user: UserEntity
+    ) {
+        this.logger.log('REST request to create tenant by user with id: ', user.id);
+        return this.tenantService.createTenant(tenantDto, user);
+    }
+
+    @Post('/invite-code/:tenantId')
+    @FeatureKeys(FeatureKey.TENANT_CREATE_ALL_INVITE_CODE)
+    async createInviteCodeByTenant(@Param('tenantId') id: Tenant['id']) {
+        this.logger.log('REST request to create tenant invite code for tenant with id: ', id);
+
+        if (await this.tenantService.getActiveInviteCodeByTenantId(id)) {
+            this.logger.error('Valid code exists for tenant with id: ', id);
+            throw new BadRequestException('Valid code exists for tenant');
+        }
+
+        return this.tenantService.createInviteCodeByTenantId(id);
+    }
+
+    @Patch('/:id')
+    @FeatureKeys(FeatureKey.TENANT_ALL_ACCESS)
+    updateTenant(
+        @Param('id', ParseUUIDPipe) id: Tenant['id'],
+        @Body(new ZodValidationPipe(updateTenantSchema)) tenantDto: UpdateTenantDto,
```

Comment:
what is returned if there's too many/less/any wrong values in body? 
We'd like bad request + info what's wrong with the body